### PR TITLE
Remove static app/context reference from App

### PR DESCRIPTION
**Pull request by skywall**

### CODE REVIEW

File: app/src/main/kotlin/cz/covid19cz/erouska/utils/DeviceInfo.kt [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+import android.content.Context
 import android.os.Build
 import com.jaredrummler.android.device.DeviceName
-import cz.covid19cz.erouska.App
+import org.koin.core.KoinComponent
+import org.koin.core.get
 import java.util.*
 
-object DeviceInfo {
+object DeviceInfo : KoinComponent {
```
[the remark this answers is not in the document]
Comment (skywall) [reply app/src/main/kotlin/cz/covid19cz/erouska/utils/DeviceInfo.kt:11]: From my perspective, I'd rather turn all of this helper/manager classes into the classes instead of objects, and inject all of the dependencies properly via a constructor. If you're ok with that change, I'm ready to do it. I tried to avoid some significant changes in the current codebase.

In this specific case, I like your first suggestion, but objects don't have constructors, so I can't use constructor injection. A second suggestion is simple, but since `DeviceInfo` is used in `ViewModel`, I would prefer not to pass `Context` as an argument and leave VM without Android dependencies.

[the remark this answers is not in the document]
Comment (simekadam) [reply app/src/main/kotlin/cz/covid19cz/erouska/utils/DeviceInfo.kt:11]: Let's just convert both objects into classes with constructor injection?

[the remark this answers is not in the document]
Comment (skywall) [reply app/src/main/kotlin/cz/covid19cz/erouska/utils/DeviceInfo.kt]: @simekadam updated.

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/contacts/ContactsFragment.kt [hunk 2]
```diff
@@ -41,14 +44,14 @@ class ContactsFragment : BaseFragment<FragmentPermissionssDisabledBinding, Conta
...
     }
 
     private fun openFaq() {
-        showWeb(viewModel.getFaqUrl())
+        showWeb(viewModel.getFaqUrl(), customTabHelper)
     }
 
     private fun goToWeb() {
-        showWeb(viewModel.getHomepageLink())
+        showWeb(viewModel.getHomepageLink(), customTabHelper)
```
Comment (simekadam) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/contacts/ContactsFragment.kt:55]: It would be actually nice to refactor this so that we call it rather like `customTabHelper.showWeb(link)` but I'd leave that for a separate PR.

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt [hunk 3]
```diff
@@ -111,12 +113,8 @@ class MainActivity :
 
     override fun onStart() {
         super.onStart()
-        if (CustomTabHelper.chromePackageName != null) {
-            CustomTabsClient.bindCustomTabsService(
-                this,
-                CustomTabHelper.chromePackageName,
-                customTabsConnection
-            )
+        customTabHelper.chromePackageName?.let {
```
Comment (simekadam) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt:116]: 👍

**Comment by davidvavra** [comment]
OK to merge @simekadam ?

**Comment by simekadam** [comment]
I just wanted to build it and briefly check. Code looks good. You can do that and merge if you want. I'm afk now
